edx_api/grades/models.py

```diff
@@ -1,23 +1,69 @@
 """
 Business objects for the Grades API
 """
-from collections import Iterable
 
-from six import python_2_unicode_compatible
+from __future__ import unicode_literals
+from six import PY2, python_2_unicode_compatible
 
+# pylint: disable=no-name-in-module, import-error
+if PY2:
+    from collections import Iterable
+else:
+    from collections.abc import Iterable
+# pylint: enable=no-name-in-module, import-error
 # pylint: disable=too-few-public-methods
 
 
-@python_2_unicode_compatible
 class CurrentGrades(object):
     """
     Current Grades object representation
     """
+    @property
+    def all_current_grades(self):
+        """Helper property to return all the CurrentGrade objects"""
+        return self.current_grades.values()  # pylint: disable=no-member
+
+
+@python_2_unicode_compatible
+class CurrentGradesByCourse(CurrentGrades):
+    """
+    Represents the current grades for a specific course
+    """
     def __init__(self, current_grade_list):
         if not isinstance(current_grade_list, Iterable):
-            raise TypeError('CurrentGrades needs a Iterable object')
+            raise TypeError('CurrentGrades needs an Iterable object')
+        self.course_id = None
         self.current_grades = {}
+        for current_grade in current_grade_list:
+            if not isinstance(current_grade, CurrentGrade):
+                raise ValueError("Only CurrentGrade objects are allowed")
+            if self.course_id is None:
+                self.course_id = current_grade.course_id
+            if self.course_id is not None and current_grade.course_id != self.course_id:
+                raise ValueError("Only CurrentGrade objects for the same course are allowed")
+            self.current_grades[current_grade.username] = current_grade
+
+    def __str__(self):
+        return "<Current Grades for course {course_id}>".format(
+            course_id=self.course_id
+        )
+
+    @property
+    def all_usernames(self):
+        """Helper property to return all the usernames of the current grades"""
+        return self.current_grades.keys()
```

Comment:
Nit: as this is the same as `all_course_ids` method, could we move this to a superclass?

Reply:
I left this here because for CourseGradesByUser it doesn't quite mean the same thing, as there's only one user. The function definition is the same, but the property name is the more important part here, to have an unambigous interface to the keys.

Reply:
Ah, that makes sense. Thanks for explanation!
Btw. you can mark it as resolved, because I don't have proper permissions for this.